compiler/modules/front/symbols.bal

```diff
@@ -11,6 +11,12 @@ type ModuleSymbols record {|
     map<Import>[] partPrefixes = [];
     t:Context tc;
     boolean allowAllTypes = false;
+    map<DeferredEmptinessCheck[]> deferredEmptinessChecks = {};
```

Comment:
Why does this need to be a map rather than an array?

Reply:
Seperated out what I need to check if an empty atomic type is a recursive type into `recursiveEmptinessChecked`.

Reply:
I need this to separate cases where we have an empty atomic type that is recursive (infinite) and empty atomic type that is not recursive (something I don't think should be possible).